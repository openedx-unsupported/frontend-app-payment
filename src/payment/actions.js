import { utils } from '../common';

const { AsyncActionType } = utils;

export const FETCH_BASKET = new AsyncActionType('PAYMENT', 'FETCH_BASKET');


export const fetchBasket = () => ({
  type: FETCH_BASKET.BASE,
  payload: {},
});

export const fetchBasketBegin = () => ({
  type: FETCH_BASKET.BEGIN,
});

export const fetchBasketSuccess = result => ({
  type: FETCH_BASKET.SUCCESS,
  payload: result,
});

export const fetchBasketFailure = () => ({
  type: FETCH_BASKET.FAILURE,
});
