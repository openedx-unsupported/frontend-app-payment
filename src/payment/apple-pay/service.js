/* eslint-disable no-console */
import pick from 'lodash.pick';
import { applyConfiguration } from '../../common/serviceUtils';

let apiClient = null;

let config = {
  ECOMMERCE_BASE_URL: null,
  ECOMMERCE_RECEIPT_BASE_URL: null,
  APPLE_PAY_MERCHANT_IDENTIFIER: null,
  APPLE_PAY_MERCHANT_NAME: null,
  APPLE_PAY_COUNTRY_CODE: null,
  APPLE_PAY_CURRENCY_CODE: null,
  APPLE_PAY_START_SESSION_URL: null,
  APPLE_PAY_AUTHORIZE_URL: null,
  APPLE_PAY_SUPPORTED_NETWORKS: null,
  APPLE_PAY_MERCHANT_CAPABILITIES: null,
};

export function configureApiService(_config, _apiClient) {
  applyConfiguration(config, _config);
  config = pick(_config, Object.keys(config));
  apiClient = _apiClient;
}

/* istanbul ignore next */
const tempLogAxiosError = (error) => {
  if (error.response) {
    // The request was made and the server responded with a status code
    // that falls out of the range of 2xx
    console.log(error.response.data);
    console.log(error.response.data.statusMessage);
    console.log(error.response.status);
    console.log(error.response.headers);
  } else if (error.request) {
    // The request was made but no response was received
    // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
    // http.ClientRequest in node.js
    console.log(error.request);
  } else {
    // Something happened in setting up the request that triggered an Error
    console.log('Error', error.message);
  }
  console.log(error.config);
};


export const performApplePayPayment = ({
  totalAmount,
  onPaymentBegin,
  onPaymentComplete,
  onPaymentAuthorizationFailure,
  onMerchantValidationFailure,
  onPaymentCancel,
}) => {
  // Session Set Up
  // ------------------------

  const version = 2;
  const paymentRequest = {
    countryCode: config.APPLE_PAY_COUNTRY_CODE,
    currencyCode: config.APPLE_PAY_CURRENCY_CODE,
    supportedNetworks: config.APPLE_PAY_SUPPORTED_NETWORKS,
    merchantCapabilities: config.APPLE_PAY_MERCHANT_CAPABILITIES,
    total: {
      label: config.APPLE_PAY_MERCHANT_NAME,
      type: 'final',
      amount: totalAmount,
    },
    requiredBillingContactFields: ['postalAddress'],
  };

  const applePaySession = new global.ApplePaySession(version, paymentRequest);


  // Set Up Event Handlers on Payment Session
  // ------------------------

  applePaySession.onvalidatemerchant = (event) => {
    console.log('Validating merchant...');

    const postData = {
      url: event.validationURL,
      is_payment_microfrontend: true,
    };

    return apiClient.post(config.APPLE_PAY_START_SESSION_URL, postData)
      .then((response) => {
        console.log('Merchant validation succeeded.', response.data);
        applePaySession.completeMerchantValidation(response.data);
      })
      .catch((error) => {
        tempLogAxiosError(error);
        applePaySession.abort();
        /* istanbul ignore else */
        if (onMerchantValidationFailure) onMerchantValidationFailure(error);
      });
  };

  applePaySession.onpaymentauthorized = (event) => {
    console.log('Submitting Apple Pay payment to CyberSource...');

    const postData = event.payment;

    return apiClient.post(config.APPLE_PAY_AUTHORIZE_URL, postData)
      .then(({ data }) => {
        const orderNumber = data.number;
        console.log('CyberSource successfully authorized Apple Pay payment.', data);
        applePaySession.completePayment(global.ApplePaySession.STATUS_SUCCESS);
        /* istanbul ignore else */
        if (onPaymentComplete) onPaymentComplete(orderNumber, true);
      })
      .catch((error) => {
        tempLogAxiosError(error);
        applePaySession.completePayment(global.ApplePaySession.STATUS_FAILURE);
        /* istanbul ignore else */
        if (onPaymentAuthorizationFailure) onPaymentAuthorizationFailure(error);
      });
  };

  applePaySession.oncancel = (event) => {
    console.log('Cancel', event);
    /* istanbul ignore else */
    if (onPaymentCancel) onPaymentCancel(event);
  };


  // Begin transaction
  // ------------------------
  applePaySession.begin();
  if (onPaymentBegin) onPaymentBegin();
};

/**
 * Given an order number, and an optional arg of whether to disable the back button, redirects the
 * user to the receipt page with the correct query parameters.
 *
 * @param orderNumber The number of the order
 * @param disableBackButton Whether to disable the back button on the receipt page. We default to
 * false since the receipt page is referenced in emails/etc. We only disable the back button when
 * the flow goes through the payment page.
 */
/* istanbul ignore next - you can't mock window.location.href in jsdom */
export const redirectToReceipt = (orderNumber, disableBackButton = false) => {
  const queryParams = `order_number=${orderNumber}&disable_back_button=${Number(disableBackButton)}`;
  global.location.href = `${config.ECOMMERCE_RECEIPT_BASE_URL}?${queryParams}`;
};
