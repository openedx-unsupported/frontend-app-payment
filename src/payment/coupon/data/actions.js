import { AsyncActionType } from '../../../common/utils';

export const ADD_COUPON = new AsyncActionType('PAYMENT', 'ADD_COUPON');
export const REMOVE_COUPON = new AsyncActionType('PAYMENT', 'REMOVE_COUPON');
export const UPDATE_COUPON_DRAFT = 'PAYMENT__UPDATE_COUPON_DRAFT';

// ADD COUPON

export const addCoupon = code => ({
  type: ADD_COUPON.BASE,
  payload: {
    code,
  },
});

export const addCouponBegin = () => ({
  type: ADD_COUPON.BEGIN,
});

export const addCouponSuccess = (voucherId, code, benefit) => ({
  type: ADD_COUPON.SUCCESS,
  payload: {
    benefit,
    code,
    voucherId,
  },
});

export const addCouponFailure = errorCode => ({
  type: ADD_COUPON.FAILURE,
  payload: {
    errorCode,
  },
});

// REMOVE COUPON

export const removeCoupon = voucherId => ({
  type: REMOVE_COUPON.BASE,
  payload: {
    voucherId,
  },
});

export const removeCouponBegin = () => ({
  type: REMOVE_COUPON.BEGIN,
});

export const removeCouponSuccess = result => ({
  type: REMOVE_COUPON.SUCCESS,
  payload: {
    result,
  },
});

export const removeCouponFailure = errorCode => ({
  type: REMOVE_COUPON.FAILURE,
  payload: {
    errorCode,
  },
});

// UPDATE COUPON DRAFT

export const updateCouponDraft = code => ({
  type: UPDATE_COUPON_DRAFT,
  payload: {
    code,
  },
});
