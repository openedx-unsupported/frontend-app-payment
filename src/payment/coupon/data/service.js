import pick from 'lodash.pick';

import { handleRequestError, applyConfiguration } from '../../../common/serviceUtils';

let config = {
  ECOMMERCE_BASE_URL: null,
};

let apiClient = null; // eslint-disable-line no-unused-vars

export function configureApiService(newConfig, newApiClient) {
  applyConfiguration(config, newConfig);
  config = pick(newConfig, Object.keys(config));
  apiClient = newApiClient;
}

export async function postCoupon(code) {
  const response = await apiClient
    .post(
      `${config.ECOMMERCE_BASE_URL}/bff/payment/v0/vouchers/`,
      { code },
      {
        headers: { 'Content-Type': 'application/json' },
      },
    )
    .catch(handleRequestError);

  return response.data;
}

export async function deleteCoupon(voucherId) {
  const response = await apiClient
    .delete(`${config.ECOMMERCE_BASE_URL}/bff/payment/v0/vouchers/${voucherId}`)
    .catch(handleRequestError);

  return response.data;
}
