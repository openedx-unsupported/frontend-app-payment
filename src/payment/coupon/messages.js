import { defineMessages } from '@edx/frontend-i18n';

const messages = defineMessages({
  'payment.coupon.label': {
    id: 'payment.coupon.label',
    defaultMessage: 'Add coupon code (optional)',
    description: 'Label for the add coupon form',
  },
  'payment.coupon.submit': {
    id: 'payment.coupon.submit',
    defaultMessage: 'Apply',
    description: 'Submit button for the add coupon form',
  },
  'payment.coupon.remove': {
    id: 'payment.coupon.remove',
    defaultMessage: 'Remove',
    description: 'Submit button to remove a coupon',
  },
  'payment.coupon.benefit_value': {
    id: 'payment.coupon.benefit_value',
    defaultMessage: 'Coupon {code} applied for {value} off',
    description: 'The description of a coupon for a percentage off the total price. Percentage symbol is included in {value}',
  },
  'payment.coupon.benefit.default': {
    id: 'payment.coupon.benefit.default',
    defaultMessage: 'Coupon {code} applied',
    description: 'The description of a coupon without any specific monetary value off.',
  },
});

export default messages;
