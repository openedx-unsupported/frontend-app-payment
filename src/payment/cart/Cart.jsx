import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormattedMessage, injectIntl, intlShape } from '@edx/frontend-platform/i18n';

import messages from './Cart.messages';
import { cartSelector } from '../data/selectors';
import { ORDER_TYPES } from '../data/constants';

import BulkOrderSummaryTable from './BulkOrderSummaryTable';
import CartSkeleton from './CartSkeleton';
import CartContents from './CartContents';
import CouponForm from './CouponForm';
import CurrencyDisclaimer from './CurrencyDisclaimer';
import OrderSummary from './OrderSummary';
import OrderDetails from './order-details';
import Offers from './Offers';
import ProductLineItem from './ProductLineItem';
import SummaryTable from './SummaryTable';
import TotalTable from './TotalTable';
import UpdateQuantityForm from './UpdateQuantityForm';

class Cart extends React.Component {
  renderCart() {
    const {
      products,
      orderType,
      isCurrencyConverted,
      isPaymentVisualExperiment,
      isNumEnrolledExperiment,
      REV1045Experiment,
      isPriceMessageExperiment,
      enrollmentCountData,
      orderTotal,
      showCouponForm,
      summaryPrice,
      summarySubtotal,
      summaryQuantity,
      summaryDiscounts,
      offers,
      loaded
    } = this.props;

    const isBulkOrder = orderType === ORDER_TYPES.BULK_ENROLLMENT;

    return (
      <div>
        <span className="sr-only">
          <FormattedMessage
            id="payment.screen.reader.cart.details.loaded"
            defaultMessage="Shopping cart details are loaded."
            description="Screen reader text to be read when cart details load."
          />
        </span>

        <CartContents>
          {products && products.length ? products.map(product => (<ProductLineItem
            key={product.title}
            isNumEnrolledExperiment={isNumEnrolledExperiment}
            REV1045Experiment={REV1045Experiment}
            isPaymentVisualExperiment={isPaymentVisualExperiment}
            enrollmentCountData={enrollmentCountData}
            {...product}
          />)) : (REV1045Experiment ? <div className="row align-items-center mb-5">
        <div className="col-5">
          <div className="skeleton embed-responsive embed-responsive-16by9" />
        </div>
        <div className="col-7">
          <div className="skeleton py-2 mb-3 w-50" />
          <div className="skeleton py-2 mr-4" />
        </div>
      </div> : null)}

          {isBulkOrder ? <UpdateQuantityForm /> : null}
        </CartContents>

        {!REV1045Experiment || loaded ? <OrderSummary>
          {isBulkOrder ? (
            <BulkOrderSummaryTable
              price={summaryPrice}
              subtotal={summarySubtotal}
              quantity={summaryQuantity}
            />
          ) : (
            <SummaryTable price={summaryPrice} />
          )}

          <Offers discounts={summaryDiscounts} offers={offers} isBundle={products.length > 1} />
          {showCouponForm ?
            <CouponForm isPaymentVisualExperiment={isPaymentVisualExperiment} /> : null}
          <TotalTable total={orderTotal} />
          {isCurrencyConverted ? <CurrencyDisclaimer /> : null}
        </OrderSummary> : (<React.Fragment><div className="skeleton py-2 mb-3 w-50" />
      <div className="skeleton py-2 mb-2" />
      <div className="skeleton py-2 mb-5" /></React.Fragment>)}

        <OrderDetails REV1045Experiment={REV1045Experiment} />
        {isPriceMessageExperiment ?
          <div className="price-messaging">
            <h5>Thanks for supporting our mission!</h5>
            <p>Did you know? As a non-profit, we rely on paid commitments to support our mission of increasing access to high-quality education for learners like you, everywhere.</p>
          </div>
        : null}
      </div>
    );
  }

  render() {
    const {
      intl,
      loading,
      loaded,
      REV1045Experiment,
    } = this.props;

    return (
      <section
        aria-live="polite"
        aria-relevant="all"
        aria-label={intl.formatMessage(messages['payment.section.cart.label'])}
      >
        {!REV1045Experiment && loading ? <CartSkeleton /> : this.renderCart() }
      </section>
    );
  }
}


Cart.propTypes = {
  intl: intlShape.isRequired,
  isPaymentVisualExperiment: PropTypes.bool,
  isNumEnrolledExperiment: PropTypes.bool,
  REV1045Experiment: PropTypes.bool,
  isPriceMessageExperiment: PropTypes.bool,
  enrollmentCountData: PropTypes.arrayOf(PropTypes.shape({
    key: PropTypes.string,
    enrollment_count: PropTypes.number,
  })),
  loading: PropTypes.bool,
  loaded: PropTypes.bool,
  products: PropTypes.arrayOf(PropTypes.shape({
    imageUrl: PropTypes.string,
    title: PropTypes.string,
    certificateType: PropTypes.oneOf(['audit', 'honor', 'verified', 'no-id-professional', 'professional', 'credit']),
  })),
  showCouponForm: PropTypes.bool,
  orderType: PropTypes.oneOf(Object.values(ORDER_TYPES)),
  isCurrencyConverted: PropTypes.bool,
  orderTotal: PropTypes.number,
  offers: PropTypes.arrayOf(PropTypes.shape({
    benefitType: PropTypes.oneOf(['Percentage', 'Absolute']).isRequired,
    benefitValue: PropTypes.number.isRequired,
    provider: PropTypes.string,
  })),
  summarySubtotal: PropTypes.number,
  summaryQuantity: PropTypes.number,
  summaryDiscounts: PropTypes.number,
  summaryPrice: PropTypes.number,
};

Cart.defaultProps = {
  isPaymentVisualExperiment: false,
  isNumEnrolledExperiment: false,
  REV1045Experiment: false,
  isPriceMessageExperiment: false,
  enrollmentCountData: null,
  loading: true,
  loaded: false,
  products: [],
  orderType: ORDER_TYPES.SEAT,
  showCouponForm: false,
  isCurrencyConverted: false,
  orderTotal: undefined,
  offers: [],
  summarySubtotal: undefined,
  summaryQuantity: undefined,
  summaryDiscounts: undefined,
  summaryPrice: undefined,
};

export default connect(cartSelector)(injectIntl(Cart));
