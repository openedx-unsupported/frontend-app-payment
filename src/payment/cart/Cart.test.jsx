/* eslint-disable global-require */
import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import { IntlProvider, configure as configureI18n } from '@edx/frontend-i18n';
import { fetchUserAccountSuccess } from '@edx/frontend-auth';
import { Factory } from 'rosie';
import { createStore } from 'redux';

import '../__factories__/basket.factory';
import '../../__factories__/configuration.factory';
import '../../__factories__/userAccount.factory';
import { configuration } from '../../environment';
import messages from '../../i18n';
import Cart from './Cart';
import createRootReducer from '../../data/reducers';
import { fetchBasket, basketDataReceived } from '../data/actions';
import { transformResults } from '../data/service';

configureI18n(configuration, messages);

describe('<Cart />', () => {
  let store;
  let initialState;
  let tree;
  let userAccount;

  beforeEach(() => {
    userAccount = Factory.build('userAccount');
    initialState = {
      configuration: Factory.build('configuration'),
      authentication: {
        userId: 9,
        username: userAccount.username,
      },
    };

    store = createStore(createRootReducer(), initialState);
    store.dispatch(fetchUserAccountSuccess(userAccount));

    const component = (
      <IntlProvider locale="en">
        <Provider store={store}>
          <Cart />
        </Provider>
      </IntlProvider>
    );
    tree = renderer.create(component);
  });

  it('renders the loading skeleton', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders a basic, one product cart', () => {
    act(() => {
      store.dispatch(basketDataReceived(transformResults(Factory.build(
        'basket',
        {},
        { numProducts: 1 },
      ))));
      store.dispatch(fetchBasket.fulfill());
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders a basic, one product cart with coupon form', () => {
    act(() => {
      store.dispatch(basketDataReceived(transformResults(Factory.build(
        'basket',
        { show_coupon_form: true },
        { numProducts: 1 },
      ))));
      store.dispatch(fetchBasket.fulfill());
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders a cart with an offer', () => {
    act(() => {
      store.dispatch(basketDataReceived(transformResults(Factory.build(
        'basket',
        {
          offers: [{ benefitValue: 50, benefitType: 'Percentage', provider: 'Pied Piper' }],
        },
        { numProducts: 1 },
      ))));
      store.dispatch(fetchBasket.fulfill());
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders a cart in non USD currency', () => {
    // This test uses its own setup since we don't have actions to update currency.
    store = createStore(
      createRootReducer(),
      Object.assign({}, initialState, {
        payment: {
          currency: {
            currencyCode: 'MXN',
            conversionRate: 19.092733,
          },
        },
      }),
    );
    const component = (
      <IntlProvider locale="en">
        <Provider store={store}>
          <Cart />
        </Provider>
      </IntlProvider>
    );
    tree = renderer.create(component);

    act(() => {
      store.dispatch(fetchUserAccountSuccess(userAccount));
      store.dispatch(basketDataReceived(transformResults(Factory.build(
        'basket',
        {},
        { numProducts: 1 },
      ))));
      store.dispatch(fetchBasket.fulfill());
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders a bulk enrollment cart', () => {
    act(() => {
      store.dispatch(fetchUserAccountSuccess(userAccount));
      store.dispatch(basketDataReceived(transformResults(Factory.build(
        'basket',
        {},
        { numProducts: 1, productType: 'Enrollment Code' },
      ))));
      store.dispatch(fetchBasket.fulfill());
    });

    expect(tree).toMatchSnapshot();
  });
});
