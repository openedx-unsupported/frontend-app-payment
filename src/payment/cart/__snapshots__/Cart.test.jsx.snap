// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> renders a basic, one product cart 1`] = `
<section
  aria-label="Shopping Cart Details"
  aria-live="polite"
  aria-relevant="all"
>
  <div>
    <span
      className="sr-only"
    >
      <span>
        Shopping cart details are loaded.
      </span>
    </span>
    <div
      className="basket-section"
    >
      <h5
        aria-level="2"
      >
        <span>
          In Your Cart
        </span>
      </h5>
      <p>
        <span>
          Your purchase contains the following:
        </span>
      </p>
      <div
        className="row align-items-center mb-3"
      >
        <div
          className="col-5"
        >
          <div
            className="embed-responsive embed-responsive-16by9"
          >
            <img
              alt=""
              className="img-thumbnail product-thumbnail"
              src="https://edx.org/vot.png"
            />
          </div>
        </div>
        <div
          className="col-7"
        >
          <h6
            aria-level="3"
            className="m-0"
          >
            Introduction to Llamass
          </h6>
          <p
            className="m-0"
          >
            <span>
              Verified Certificate
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="summary-heading"
      aria-live="polite"
      className="basket-section"
      role="region"
    >
      <h5
        aria-level="2"
        id="summary-heading"
      >
        <span>
          Summary
        </span>
      </h5>
       
      <div
        className="summary-row d-flex"
      >
        <span
          className="flex-grow-1"
        >
          <span>
            Price
          </span>
        </span>
        <span>
          <span>
            $464.88
          </span>
        </span>
      </div>
      <div
        className="summary-row font-weight-bold d-flex"
      >
        <span
          className="flex-grow-1"
        >
          <span>
            TOTAL
          </span>
        </span>
        <span
          className="text-right"
        >
          <span>
            $464.88
          </span>
        </span>
      </div>
    </div>
    <div
      className="basket-section"
    >
      <h5
        aria-level="2"
      >
        Order Details
      </h5>
      <p>
        After you complete your order you will be able to select course dates from your dashboard.
      </p>
    </div>
  </div>
</section>
`;

exports[`<Cart /> renders a basic, one product cart with coupon form 1`] = `
<section
  aria-label="Shopping Cart Details"
  aria-live="polite"
  aria-relevant="all"
>
  <div>
    <span
      className="sr-only"
    >
      <span>
        Shopping cart details are loaded.
      </span>
    </span>
    <div
      className="basket-section"
    >
      <h5
        aria-level="2"
      >
        <span>
          In Your Cart
        </span>
      </h5>
      <p>
        <span>
          Your purchase contains the following:
        </span>
      </p>
      <div
        className="row align-items-center mb-3"
      >
        <div
          className="col-5"
        >
          <div
            className="embed-responsive embed-responsive-16by9"
          >
            <img
              alt=""
              className="img-thumbnail product-thumbnail"
              src="https://edx.org/watpewal.gif"
            />
          </div>
        </div>
        <div
          className="col-7"
        >
          <h6
            aria-level="3"
            className="m-0"
          >
            Introduction to Pigeonss
          </h6>
          <p
            className="m-0"
          >
            <span>
              Verified Certificate
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="summary-heading"
      aria-live="polite"
      className="basket-section"
      role="region"
    >
      <h5
        aria-level="2"
        id="summary-heading"
      >
        <span>
          Summary
        </span>
      </h5>
       
      <div
        className="summary-row d-flex"
      >
        <span
          className="flex-grow-1"
        >
          <span>
            Price
          </span>
        </span>
        <span>
          <span>
            $445.19
          </span>
        </span>
      </div>
      <form
        className="summary-row d-flex align-items-end"
        onSubmit={[Function]}
      >
        <div
          className="form-group mb-0 mr-2"
        >
          <label
            className="h6 d-block"
            htmlFor="couponField"
          >
            <span>
              Add coupon code (optional)
            </span>
          </label>
          <input
            aria-describedby=""
            className="form-control"
            defaultValue=""
            id="couponField"
            name="couponField"
            type="text"
          />
        </div>
        <button
          className="btn btn-outline-primary"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          <span>
            Apply
          </span>
        </button>
      </form>
      <div
        className="summary-row font-weight-bold d-flex"
      >
        <span
          className="flex-grow-1"
        >
          <span>
            TOTAL
          </span>
        </span>
        <span
          className="text-right"
        >
          <span>
            $445.19
          </span>
        </span>
      </div>
    </div>
    <div
      className="basket-section"
    >
      <h5
        aria-level="2"
      >
        Order Details
      </h5>
      <p>
        After you complete your order you will be able to select course dates from your dashboard.
      </p>
    </div>
  </div>
</section>
`;

exports[`<Cart /> renders a bulk enrollment cart 1`] = `
<section
  aria-label="Shopping Cart Details"
  aria-live="polite"
  aria-relevant="all"
>
  <div>
    <span
      className="sr-only"
    >
      <span>
        Shopping cart details are loaded.
      </span>
    </span>
    <div
      className="basket-section"
    >
      <h5
        aria-level="2"
      >
        <span>
          In Your Cart
        </span>
      </h5>
      <p>
        <span>
          Your purchase contains the following:
        </span>
      </p>
      <div
        className="row align-items-center mb-3"
      >
        <div
          className="col-5"
        >
          <div
            className="embed-responsive embed-responsive-16by9"
          >
            <img
              alt=""
              className="img-thumbnail product-thumbnail"
              src="https://edx.org/eckifega.png"
            />
          </div>
        </div>
        <div
          className="col-7"
        >
          <h6
            aria-level="3"
            className="m-0"
          >
            Introduction to Chickenss
          </h6>
          <p
            className="m-0"
          >
            <span>
              Verified Certificate
            </span>
          </p>
        </div>
      </div>
      <form
        className="summary-row form-inline"
        onSubmit={[Function]}
      >
        <div
          className="form-group mr-2"
        >
          <label
            htmlFor="code-quantity"
          >
            <span>
              Quantity
            </span>
          </label>
          <div
            className="position-relative"
          >
            <input
              className="form-control form-control-sm"
              defaultValue={1}
              id="code-quantity"
              max="100"
              min="1"
              name="code-quantity"
              style={
                Object {
                  "width": "5rem",
                }
              }
              type="number"
            />
            <small
              className="text-muted small"
              style={
                Object {
                  "left": 0,
                  "position": "absolute",
                  "top": "100%",
                }
              }
            >
              <span>
                Max: 100
              </span>
            </small>
          </div>
        </div>
        <button
          aria-disabled={false}
          aria-live="assertive"
          className="pgn__stateful-btn pgn__stateful-btn-state-default btn btn-primary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          <span
            className="d-flex align-items-center justify-content-center"
          >
            <span>
              <span>
                Update
              </span>
            </span>
          </span>
        </button>
      </form>
    </div>
    <div
      aria-labelledby="summary-heading"
      aria-live="polite"
      className="basket-section"
      role="region"
    >
      <h5
        aria-level="2"
        id="summary-heading"
      >
        <span>
          Summary
        </span>
      </h5>
       
      <div
        className="summary-row"
      >
        <table
          className="w-100"
        >
          <thead>
            <tr>
              <th
                className="font-weight-normal"
              >
                <span>
                  Price
                </span>
              </th>
              <th
                className="font-weight-normal"
              >
                <span>
                  Quantity
                </span>
              </th>
              <th
                className="font-weight-normal text-right"
              >
                <span>
                  Subtotal
                </span>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <span>
                  $92.77
                </span>
              </td>
              <td>
                1
              </td>
              <td
                className="text-right"
              >
                <span>
                  $92.77
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="summary-row font-weight-bold d-flex"
      >
        <span
          className="flex-grow-1"
        >
          <span>
            TOTAL
          </span>
        </span>
        <span
          className="text-right"
        >
          <span>
            $92.77
          </span>
        </span>
      </div>
    </div>
    <div
      className="basket-section"
    >
      <h5
        aria-level="2"
      >
        Order Details
      </h5>
      <p>
        By purchasing, you and your organization agree to the following terms:
      </p>
      <ul>
        <li>
          Each code is valid for the one course covered and can be used only one time.
        </li>
        <li>
          You are responsible for distributing codes to your learners in your organization.
        </li>
        <li>
          Each code will expire in one year from date of purchase or, if earlier, once the course is closed.
        </li>
        <li>
          If a course is not designated as self-paced, you should confirm that a course run is available before expiration.
        </li>
        <li>
          You may not resell codes to third parties.
        </li>
        <li>
          All edX for Business sales are final and not eligible for refunds.
        </li>
      </ul>
      <p>
        You will receive an email at person@edx.org with your enrollment code(s).
      </p>
    </div>
  </div>
</section>
`;

exports[`<Cart /> renders a cart in non USD currency 1`] = `
<section
  aria-label="Shopping Cart Details"
  aria-live="polite"
  aria-relevant="all"
>
  <div>
    <span
      className="sr-only"
    >
      <span>
        Shopping cart details are loaded.
      </span>
    </span>
    <div
      className="basket-section"
    >
      <h5
        aria-level="2"
      >
        <span>
          In Your Cart
        </span>
      </h5>
      <p>
        <span>
          Your purchase contains the following:
        </span>
      </p>
      <div
        className="row align-items-center mb-3"
      >
        <div
          className="col-5"
        >
          <div
            className="embed-responsive embed-responsive-16by9"
          >
            <img
              alt=""
              className="img-thumbnail product-thumbnail"
              src="https://edx.org/lu.jpg"
            />
          </div>
        </div>
        <div
          className="col-7"
        >
          <h6
            aria-level="3"
            className="m-0"
          >
            Introduction to Guanacos
          </h6>
          <p
            className="m-0"
          >
            <span>
              Verified Certificate
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="summary-heading"
      aria-live="polite"
      className="basket-section"
      role="region"
    >
      <h5
        aria-level="2"
        id="summary-heading"
      >
        <span>
          Summary
        </span>
      </h5>
       
      <div
        className="summary-row d-flex"
      >
        <span
          className="flex-grow-1"
        >
          <span>
            Price
          </span>
        </span>
        <span>
          <span>
            <span>
              MX$1,264
            </span>
             *
          </span>
        </span>
      </div>
      <div
        className="summary-row font-weight-bold d-flex"
      >
        <span
          className="flex-grow-1"
        >
          <span>
            TOTAL
          </span>
        </span>
        <span
          className="text-right"
        >
          <span>
            <span>
              MX$1,264
            </span>
             *
          </span>
        </span>
      </div>
      <div
        className="text-muted font-italic"
      >
        <span>
          * This total contains an approximate conversion. You will be charged 
          <span>
            $66.22
          </span>
           USD.
        </span>
      </div>
    </div>
    <div
      className="basket-section"
    >
      <h5
        aria-level="2"
      >
        Order Details
      </h5>
      <p>
        After you complete your order you will be able to select course dates from your dashboard.
      </p>
    </div>
  </div>
</section>
`;

exports[`<Cart /> renders a cart with an offer 1`] = `
<section
  aria-label="Shopping Cart Details"
  aria-live="polite"
  aria-relevant="all"
>
  <div>
    <span
      className="sr-only"
    >
      <span>
        Shopping cart details are loaded.
      </span>
    </span>
    <div
      className="basket-section"
    >
      <h5
        aria-level="2"
      >
        <span>
          In Your Cart
        </span>
      </h5>
      <p>
        <span>
          Your purchase contains the following:
        </span>
      </p>
      <div
        className="row align-items-center mb-3"
      >
        <div
          className="col-5"
        >
          <div
            className="embed-responsive embed-responsive-16by9"
          >
            <img
              alt=""
              className="img-thumbnail product-thumbnail"
              src="https://edx.org/lekigud.gif"
            />
          </div>
        </div>
        <div
          className="col-7"
        >
          <h6
            aria-level="3"
            className="m-0"
          >
            Introduction to Pigs
          </h6>
          <p
            className="m-0"
          >
            <span>
              Verified Certificate
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="summary-heading"
      aria-live="polite"
      className="basket-section"
      role="region"
    >
      <h5
        aria-level="2"
        id="summary-heading"
      >
        <span>
          Summary
        </span>
      </h5>
       
      <div
        className="summary-row d-flex"
      >
        <span
          className="flex-grow-1"
        >
          <span>
            Price
          </span>
        </span>
        <span>
          <span>
            $158.87
          </span>
        </span>
      </div>
      <div
        className="summary-row"
      >
        <p
          className="d-flex m-0"
        >
          <span
            className="flex-grow-1"
          >
            <span>
              Discounts applied
            </span>
          </span>
          <span>
            <span>
              $0.00
            </span>
          </span>
        </p>
        <p
          className="m-0 text-muted"
        >
          <span>
            <span>
              50%
            </span>
             discount provided by Pied Piper.
          </span>
        </p>
      </div>
      <div
        className="summary-row font-weight-bold d-flex"
      >
        <span
          className="flex-grow-1"
        >
          <span>
            TOTAL
          </span>
        </span>
        <span
          className="text-right"
        >
          <span>
            $158.87
          </span>
        </span>
      </div>
    </div>
    <div
      className="basket-section"
    >
      <h5
        aria-level="2"
      >
        Order Details
      </h5>
      <p>
        After you complete your order you will be able to select course dates from your dashboard.
      </p>
    </div>
  </div>
</section>
`;

exports[`<Cart /> renders the loading skeleton 1`] = `
<section
  aria-label="Shopping Cart Details"
  aria-live="polite"
  aria-relevant="all"
>
  <div>
    <span
      className="sr-only"
    >
      <span>
        Loading, please wait...
      </span>
    </span>
    <div
      className="skeleton py-2 mb-3 w-50"
    />
    <div
      className="skeleton py-2 mb-4 mr-4"
    />
    <div
      className="row align-items-center mb-5"
    >
      <div
        className="col-5"
      >
        <div
          className="skeleton embed-responsive embed-responsive-16by9"
        />
      </div>
      <div
        className="col-7"
      >
        <div
          className="skeleton py-2 mb-3 w-50"
        />
        <div
          className="skeleton py-2 mr-4"
        />
      </div>
    </div>
    <div
      className="skeleton py-2 mb-3 w-50"
    />
    <div
      className="skeleton py-2 mb-2"
    />
    <div
      className="skeleton py-2 mb-5"
    />
    <div
      className="skeleton py-2 mb-3 w-50"
    />
    <div
      className="skeleton py-2 mb-2 mr-4"
    />
    <div
      className="skeleton py-2 mb-5 w-75"
    />
  </div>
</section>
`;
